app/controllers/candidate_interface/application_form_controller.rb

```diff
@@ -73,6 +73,13 @@ def review_submitted
       @application_form = current_application
     end
 
+    def review_previous_application
+      @application_form = ApplicationForm.find(params['id']) if params['id'].present?
```

Comment:
This needs to be scoped to the current user, otherwise everyone can see everyone elses application.

Why does it check for the presence of the ID? It will always be here, otherwise the you can't visit the route.